plugin-testlib/src/test/kotlin/io/spine/tools/gragle/testing/ReplacementTest.kt

```diff
@@ -0,0 +1,172 @@
+import io.spine.tools.gradle.testing.Replacement
+import java.io.File
+import java.nio.file.Path
+import java.security.SecureRandom
+import java.util.*
+import kotlin.collections.ArrayList
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.io.TempDir
+
+class `'Replacement' should` {
+
+    private lateinit var folder: File
+
+    @BeforeEach
+    fun setUp(@TempDir tempDir: Path) {
+        folder = tempDir.toFile()
+    }
+
+    @Test
+    fun `frame the token name with '@' symbols`() {
+        val original = "mytoken"
+        val actual = Replacement(original, "").token()
+        assertThat(actual)
+            .isEqualTo("@$original@")
+    }
+
+    @Test
+    fun `not allow empty tokens`() {
+        assertThrows<IllegalArgumentException> { Replacement("", "value")  }
+    }
+
+    @Test
+    fun `allow empty values-to-replace-with`() {
+        assertThat(Replacement("sometoken", "").value)
+            .isEmpty()
+    }
+
+    @Nested
+    inner class `replace all token occurrences` {
+
+        @Test
+        fun `in a file`() {
+            val ref = TestData(folder, TEXT)
+            val value = "replaced"
+            Replacement(TOKEN_NAME, value).replaceIn(ref.file)
+            ref.assertReplaced(TOKEN_NAME, value)
+        }
+
+        @Test
+        fun `in files residing in a folder and its subfolders`() {
+            val refs = generateFiles(folder)
+
+            val value = "recursively-replaced"
+            Replacement(TOKEN_NAME, value)
+                .replaceIn(folder)
+            for (ref in refs) {
+                ref.assertReplaced(TOKEN_NAME, value)
+            }
+        }
+
+        @Test
+        fun `in all files in the folder recursively, excluding some folder`() {
+            val toReplace = generateFiles(folder)
+
+            val excludedFolder = folder.resolve("untouchable")
+            excludedFolder.mkdirs()
+            val toExclude = generateFiles(excludedFolder)
+
+            val value = "recursively-replaced"
+            Replacement(TOKEN_NAME, value)
+                .replaceIn(folder, excludedFolder)
+
+            for (ref in toReplace) {
+                ref.assertReplaced(TOKEN_NAME, value)
+            }
+            for (ref in toExclude) {
+                ref.assertNotReplaced()
+            }
+        }
+    }
+
+    private fun generateFiles(folder: File): ArrayList<TestData> {
+        val refs = ArrayList<TestData>()
+        for (i in 0..10) {
+            refs.add(TestData(folder, TEXT))
+        }
+        return refs
+    }
+
+    companion object {
+
+        private val random: Random = SecureRandom()
+
+        const val TOKEN_NAME = "TEST_TOKEN"
+        const val TEXT = "This is a `@$TOKEN_NAME@` which should be replaced with " +
+                "`@$TOKEN_NAME@`. \n\n\n And this `@TEST@ should remain the same."
```

Comment:
If these constants are only used inside this test class, we could extract them to the file level and make them private.

I'd suggest doing the same with the `random` instance as well.